esp/services/ldapenvironment/ldapenvironmentService.cpp

```diff
@@ -142,9 +142,19 @@ bool CldapenvironmentEx::changePermissions(const char * ou, const char * userFQD
     return ok;
 }
 
-bool CldapenvironmentEx::createSecret(const char * secretName, const char * username, const char * pwd, StringBuffer & notes)
+bool CldapenvironmentEx::createSecret(SecretType type, const char * secretName, const char * username, const char * pwd, StringBuffer & notes)
 {
-    VStringBuffer cmdLineSafe("kubectl create secret generic %s --from-literal=username=%s --from-literal=password=", secretName, username);
+    StringBuffer cmdLineSafe;
+    switch (type)
+    {
+        case ST_K8S:
+            cmdLineSafe.appendf("kubectl create secret generic %s --from-literal=username=%s --from-literal=password=", secretName, username);
+            break;
+        case ST_VAULT:
+            cmdLineSafe.appendf("vault kv put secret/authn/%s username=%s password=", secretName, username);
```

Comment:
In the future we may need some flexibility around the path that it gets stored in.  I'm not sure how vault will be used in LN.  But there's no way of knowing yet so this is great for now.

Reply:
@afishbeck Would it be too much to ask the user to enter the full vault path (secret/authn/my-authn-vault) ?